src/foam/u2/search/TextSearchView.js

```diff
@@ -100,11 +108,27 @@ foam.CLASS({
       name: 'updateValue',
       code: function() {
         var value = this.view.data;
-        this.predicate = ! value ?
-            this.True.create() :
-            this.richSearch ?
-                this.queryParser.parseString(value) :
-                this.ContainsIC(this.property, value);
+
+        if ( ! value ) {
+          this.predicate = this.True.create();
+          return;
+        }
+
+        if ( this.richSearch ) {
+          this.predicate = this.filterController.filters
+            .map((name) => this.of.getAxiomByName(name))
+            .filter((property) => foam.core.String.isInstance(property))
+            .reduce(
+              (acc, property) => this.OR(
+                this.CONTAINS_IC(property, value),
+                acc || this.False.create()
+              ),
+              this.queryParser.parseString(value)
+            );
+          return;
+        }
+
+        this.predicate = this.CONTAINS_IC(this.property, value);
```

Comment:
There should be an option for turning CONTAINS_IC on or off since it requires doing a full-table scan which is okay for small reference tables but could be very expensive for tables like Transactions. There's a way to fix this by creating a new type of keyword search MDAO Index (like an inverted-index or FMIndex) but we don't have this yet. Actually, we have the start of an FMIndex.